domain/application/state/storage.go

```diff
@@ -241,48 +252,81 @@ func (st *State) insertUnitStorage(
 			}
 
 			if err := st.createUnitStorageInstance(ctx, tx, unitUUID, inst); err != nil {
-				return errors.Capture(err)
+				return result, errors.Capture(err)
 			}
 
+			result[i].instArgs[c].StorageID = storageID
+			result[i].instArgs[c].StorageUUID = instUUID
```

Comment:
Don't do it this way, it adds meaningless complexity (harder to grok). Create a slice on line 255 and then set it to `result[i].instArgs`   at the end of the for loop.

Reply:
This was dumb. Fixed.